packages/docusaurus-theme-classic/src/validateThemeConfig.js

```diff
@@ -6,6 +6,24 @@
  */
 
 const Joi = require('@hapi/joi');
+const merge = require('lodash/merge');
+
+const DEFAULT_COLOR_MODE_CONFIG = {
+  defaultMode: 'light',
+  disableSwitch: false,
+  respectPrefersColorScheme: false,
+  switchConfig: {
+    darkIcon: '🌜',
+    darkIconStyle: {},
+    lightIcon: '🌞',
+    lightIconStyle: {},
+  },
+};
+exports.DEFAULT_COLOR_MODE_CONFIG = DEFAULT_COLOR_MODE_CONFIG;
+
+exports.mergeDefault = (config, defaultConfig) => {
+  return merge({}, defaultConfig, config);
+};
```

Comment:
this method should be moved in tests, because we shouldn't need to merge defaults in production code, as Joi is supposed to do that job already